server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java

```diff
@@ -210,32 +254,48 @@ public final class MergePolicyConfig {
         );
     }
 
-    void setSegmentsPerTier(Double segmentsPerTier) {
-        mergePolicy.setSegmentsPerTier(segmentsPerTier);
+    void setMergePolicyType(Type type) {
+        this.mergePolicyType = type;
+    }
+
+    void setSegmentsPerTier(double segmentsPerTier) {
+        tieredMergePolicy.setSegmentsPerTier(segmentsPerTier);
+        logByteSizeMergePolicy.setMergeFactor((int) segmentsPerTier);
     }
 
     void setMaxMergedSegment(ByteSizeValue maxMergedSegment) {
-        mergePolicy.setMaxMergedSegmentMB(maxMergedSegment.getMbFrac());
+        tieredMergePolicy.setMaxMergedSegmentMB(maxMergedSegment.getMbFrac());
+        // Note: max merge MB has different semantics on LogByteSizeMergePolicy: it's the maximum size for a segment to be considered for a
```

Comment:
I think this opens us up for degenerate cases in both ends, either now stopping at 2.5GB or going up to 25GB. I imagine this being provoked by different input styles (either document size, frequency etc), i.e., some data streams work well (hitting around 5GB) whereas others may suffer from worse search or indexing (due to the larger merges) performance?

I wonder if we could adapt the log merge policy to be closer to the tiered merge policy here. I.e., if it can merge together 2 adjacent segments but not 3, then merge the 2?

Reply:
Thanks for your question, it made me check the actual `LogByteSizeMergePolicy` behavior, and actually my comment is not right, `LogByteSizeMergePolicy` works as you describe since may last year via https://github.com/apache/lucene/pull/935, which introduced the behavior you described as part of fixing another problem with this merge policy, the fact that this merge policy packs segments together is even tested.